### PR TITLE
Remove synthetic testing infrastructure for HF/NLP

### PR DESCRIPTION
**Pull request by dakinggg**
# What does this PR do?
This PR removes the "synthetic" testing infrastructure setup for HF models, instead using a set of simple fixtures and pytest configuration to avoid downloading things from the HF hub multiple times. These fixtures should be easier to use (they are just the expected HF objects), and do away with a lot of extra code written for setting up the synthetic infrastructure.

# What issue(s) does this change relate to?
Closes [CO-1119](https://mosaicml.atlassian.net/browse/CO-1119)

# Before submitting
- [x] Have you read the [contributor guidelines](https://github.com/mosaicml/composer/blob/dev/CONTRIBUTING.md)?
- [x] Was this change discussed/approved in a GitHub issue first? It is much more likely to be merged if so.
- [x] Did you update any related docs and document your change?
- [x] Did you update any related tests and add any new tests related to your change? (see [testing](https://github.com/mosaicml/composer/blob/dev/CONTRIBUTING.md#running-tests))
- [x] Did you run the tests locally to make sure they pass?
- [x] Did you run `pre-commit` on your change? (see the `pre-commit` section of [prerequisites](https://github.com/mosaicml/composer/blob/dev/CONTRIBUTING.md#prerequisites))

[CO-1119]: https://mosaicml.atlassian.net/browse/CO-1119?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: tests/fixtures/fixtures.py
Comment (mvpatel2000) [inline tests/fixtures/fixtures.py:177]:
```suggestion
    return tiny_gpt_model_helper(_session_tiny_gpt_config)
```
what's the point of splitting this function out? can we flatten

Comment (dakinggg) [reply tests/fixtures/fixtures.py:177]: yeah, sure. no reason

**Comment by dakinggg** [comment]
@nik-mosaic I'm thinking you'll be the main reviewer for this PR

File: composer/algorithms/gated_linear_units/gated_linear_units.py
Comment (nik-mosaic) [inline composer/algorithms/gated_linear_units/gated_linear_units.py:168]: Why was optimizers removed?

File: tests/algorithms/test_alibi.py
Comment (nik-mosaic) [inline tests/algorithms/test_alibi.py:32]: Since we no longer have any synthetic state families, maybe rename to "dummy model family"?

File: tests/common/models.py
Comment (nik-mosaic) [inline tests/common/models.py:397]: Maybe add a comment here similar to the one where pytest.tiny_gpt_model is defined, explaining why we use this, so it's clear in both places?

**Review by nik-mosaic — APPROVED** [review]
Overall, great PR. I much prefer the "create a dummy model and instantiate a state object within the test" to the synthetic versions we were using before. The refactored tests are easier to read. Also, great work cleaning up some of the other poor coding practices in our tests.
I have a couple of minor comments, but other than that, this PR is ready to merge.

**Review by hanlint — APPROVED** [review]
So much cleaner, thank you for doing this!

File: composer/algorithms/gated_linear_units/gated_linear_units.py
Comment (dakinggg) [reply composer/algorithms/gated_linear_units/gated_linear_units.py:168]: mistake, will put back

File: tests/algorithms/test_alibi.py
Comment (dakinggg) [reply tests/algorithms/test_alibi.py:32]: yup, makes sense. will do

File: tests/common/models.py
Comment (dakinggg) [reply tests/common/models.py:397]: yup, makes sense. will do

**Review by mvpatel2000 — APPROVED** [review]
Approving to unblock in case my stamp is needed for codeowners

**Comment by dakinggg** [comment]
cc @knighton @karan6181 for code owner review

**Review by knighton — APPROVED** [review]

**Pull request merged**
